mailpoet/lib/Automation/Engine/API/API.php

```diff
@@ -58,24 +58,29 @@ private function registerRoute(string $route, string $endpointClass, string $met
       return $field->toArray();
     }, $endpointClass::getRequestSchema());
 
-    $this->wordPress->registerRestRoute(self::PREFIX, $route, [
-      'methods' => $method,
-      'callback' => function (WP_REST_Request $wpRequest) use ($endpointClass, $schema) {
-        try {
+    $this->wordPress->registerRestRoute(
+      self::PREFIX,
+      $route,
+      [
+        'methods' => $method,
+        'callback' => function (WP_REST_Request $wpRequest) use ($endpointClass, $schema) {
+          try {
+            $endpoint = $this->endpointContainer->get($endpointClass);
+            $wpRequest = $this->sanitizeUnknownParams($wpRequest, $schema);
+            $request = new Request($wpRequest);
+            return $endpoint->handle($request);
+          } catch (Throwable $e) {
+            return $this->convertToErrorResponse($e);
+          }
+        },
+        'permission_callback' => function () use ($endpointClass) {
           $endpoint = $this->endpointContainer->get($endpointClass);
-          $wpRequest = $this->sanitizeUnknownParams($wpRequest, $schema);
-          $request = new Request($wpRequest);
-          return $endpoint->handle($request);
-        } catch (Throwable $e) {
-          return $this->convertToErrorResponse($e);
-        }
-      },
-      'permission_callback' => function () use ($endpointClass) {
-        $endpoint = $this->endpointContainer->get($endpointClass);
-        return $endpoint->checkPermissions();
-      },
-      'args' => $schema,
-    ]);
+          return $endpoint->checkPermissions();
+        },
+        'args' => $schema,
+      ],
+      true // allow overriding
```

Comment:
It seems, allowing to overwrite in this way removes the free plugin `GET workflows` in favor of the the premium `POST workflows` route.

At least in my tests, setting the allow overwrite value to `false` recovers the `GET` endpoint

![image](https://user-images.githubusercontent.com/6458412/188061247-b82d65ad-ae5c-49bb-99ea-a8452cf456a6.png)

Reply:
Yep, good catch! I need to overwrite the PUT route but not the others, damn. I guess I can use some filter to replace it instead. Checking...

Reply:
Ahh, the fix is actually pretty simple! I'll drop this commit and make the premium route registrations run before the free ones!

Reply:
Thanks! Seems to do the trick 
![image](https://user-images.githubusercontent.com/6458412/188204855-96476b79-1ade-4b2b-b5d5-3cced4cb1bd7.png)